docs/src/user_docs/assembly/events.md

```diff
@@ -2,16 +2,23 @@
 
 Events interrupt VM execution for one cycle and hand control to the host. The host can read VM state and modify the advice provider. From the VM's perspective, `emit` has identical semantics to `noop` - the operand stack and registers remain unchanged.
 
-By convention, programs place a 32-bit `event_id` at the top of the stack before executing `emit`. The VM doesn't enforce this - the host reads and interprets the event ID.
+Event identifiers are field elements. Use a stable mapping: `event_id = blake3("<name>") mod p`, where `p` is the Goldilocks prime. The VM doesn’t enforce structure for stack-provided IDs, but immediate forms restrict inputs to this mapping.
```

Comment:
nit: I'm not sure if `blake3("<name>") mod p` is correct. We basically take the first element of the hash and do `mod p` on it - but is this the same as doing `mod p` on the whole hash?